packages/sdk/src/dataProtector/IExecDataProtector.ts

```diff
@@ -205,23 +209,24 @@ class IExecDataProtector {
       graphQLClient: this.graphQLClient,
     });
 
-  subscribe = (args: SubscribeParams) =>
+  subscribe = (args: SubscribeParams): Promise<SubscribeResponse> =>
     subscribe({
       ...args,
-      sharingContractAddress: this.sharingContractAddress,
-      iexec: this.iexec,
     });
 
-  getSubscribers = (args: SubscribeParams) =>
+  getSubscribers = (args: SubscribeParams): Promise<GetSubscribersResponse> =>
```

Comment:
Should we also have a `GetSubscribersParams` type?

I would say let's have a "CollectionIdParam" with `{ collectionId: number }` that we could reuse where needed.

And also a "SuccessResponse" with `{ success: true }`

Reply:
yeah i agree with that. I think it could done in an other PR to avoid many conflicts with other PR not yet merged